README.md

```diff
@@ -72,96 +73,105 @@ Adverbs are "slowly" and "noisily"
 
 The following are examples of valid sentences:
 
-* The smelly dog ran.
-* The smelly dog ran slowly.
-* The cat ate noisily.
-
-## Q2.1 (7 points)
+The smelly dog ran.
+The smelly dog ran slowly.
+The cat ate noisily.
+Q2.1 (7 points)
 
 Write the BNF (not EBNF) description for this language.
 
-## A2.1
-
-«replace this with your answer»
+A2.1
 
+<sentence> ::= "the" <adjectives> <noun> <verb> <adverb>
+<adjectives> ::= "" | <adjectives> <adjective>
+<adjective> ::= "lazy" | "smelly"
+<noun> ::= "dog" || "cat"
+<verb> ::= "ran" || "ate"
+<adverb> ::= "slowly" | "noisily" | ""
 
-## Q2.2 (5 points)
+Q2.2 (5 points)
 
 Write this grammar using EBNF with common extensions
 
-## A2.2
-
-«replace this with your answer»
-
+A2.2
 
-## Q2.3 (6 points)
+<<<<<<< HEAD
+sentence ::= "the", {<adjective>}, <noun>, <verb>, [<adverb>];
+=======
+sentence ::= "the", {}, , , [];
+>>>>>>> 07f1444fe496e18ef7278164002acd9890749777
 
-  Draw a diagram for an FSM which recognizes these sentences. Use EOI as the
-  event that occurs at the end-of-input. Start with a state named S0. Receiving
-  the word "the" in that state will transition to state S1. Name the final state
-  END.
+Q2.3 (6 points)
 
-  You can hand draw it and snap a picture, or use a drawing tool. Either way,
-  upload the image, and put the file name in the answer below.
+Draw a diagram for an FSM which recognizes these sentences. Use EOI as the event that occurs at the end-of-input. Start with a state named S0. Receiving the word "the" in that state will transition to state S1. Name the final state END.
 
-  (Hint: my answer has seven states including the start and end states)
+You can hand draw it and snap a picture, or use a drawing tool. Either way, upload the image, and put the file name in the answer below.
 
+(Hint: my answer has seven states including the start and end states)
 
-## A2.3
+A2.3
 
-«replace this with your answer»
+Photo is attached
```

Comment:
This works, but you have an extra state. You can merge S1 and S2:

![image](https://user-images.githubusercontent.com/10648/78731301-fc6c2f80-7904-11ea-895c-b948e131db2f.png)

# -2